### ISSUE TITLE
Concurrent persisiting and failing an received snapshot

### ISSUE THREAD

**Issue by Zelldon** [zeebe-io/zeebe#5611]
**Describe the bug**

It seems that during receiving an snapshot it can happen that concurrent the snapshot can be marked as valid and also be marked as failed, at least the logs indicates that.

```
D 2020-10-16T01:03:29.180815Z Broker-0-StateReplication-1 Waiting for more snapshot chunks of snapshot 47799-176-1602810003886, currently have 154/155  Broker-0-StateReplication-1
D 2020-10-16T01:03:29.185129Z Broker-0-StateReplication-1 Received all snapshot chunks (155/47799-176-1602810003886), snapshot 155 is valid  Broker-0-StateReplication-1
D 2020-10-16T01:03:56.833621Z Broker-0-StateReplication-1 Abort snapshot 47799-176-1602810003886 and mark it as invalid.  Broker-0-StateReplication-1
E 2020-10-16T01:03:56.941701Z raft-server-0-raft-partition-partition-1 Failed to delete not completed (orphaned) snapshot /usr/local/zeebe/data/raft-partition/partitions/1/pending/47799-176-1602810003886-1  raft-server-0-raft-partition-partition-1
D 2020-10-16T01:03:57.190593Z Broker-0-zb-actors-0 Partition role transitioning from FOLLOWER to CANDIDATE  Broker-0-zb-actors-0
W 2020-10-16T01:03:57.230198Z Broker-0-StateReplication-1 Unexpected error on persisting received snapshot.  Broker-0-StateReplication-1
D 2020-10-16T01:03:57.237029Z Broker-0-StateReplication-1 Failed to mark snapshot 47799-176-1602810003886 as valid  Broker-0-StateReplication-1
```

Found via Error reporting tool, related error group https://console.cloud.google.com/errors/CJuimPeqsIjLPw?service=zeebe&time=P7D&project=camunda-cloud-240911&authuser=1

**To Reproduce**
IDK

**Expected behavior**
No concurrency.

**Log/Stacktrace**
If possible add the full stacktrace or Zeebe log which contains the issue. 

<details><summary>Full Stacktrace</summary>
 <p>

```
java.nio.file.NoSuchFileException: /usr/local/zeebe/data/raft-partition/partitions/1/pending/388199-3996-1600963592647-5/016831.sst
at sun.nio.fs.UnixException.translateToIOException (Unknown Source)
at sun.nio.fs.UnixException.rethrowAsIOException (Unknown Source)
at sun.nio.fs.UnixException.rethrowAsIOException (Unknown Source)
at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes (Unknown Source)
at sun.nio.fs.UnixFileSystemProvider.readAttributes (Unknown Source)
at sun.nio.fs.LinuxFileSystemProvider.readAttributes (Unknown Source)
at java.nio.file.Files.readAttributes (Unknown Source)
at java.nio.file.FileTreeWalker.getAttributes (Unknown Source)
at java.nio.file.FileTreeWalker.visit (Unknown Source)
at java.nio.file.FileTreeWalker.next (Unknown Source)
at java.nio.file.Files.walkFileTree (Unknown Source)
at java.nio.file.Files.walkFileTree (Unknown Source)
at io.zeebe.util.FileUtil.deleteFolder (FileUtil.java:54)
at io.atomix.raft.snapshot.impl.FileBasedSnapshotStore.purgePendingSnapshot (FileBasedSnapshotStore.java:304)
at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (Unknown Source)
at java.util.stream.ReferencePipeline$2$1.accept (Unknown Source)
at java.util.Iterator.forEachRemaining (Unknown Source)
at java.util.Spliterators$IteratorSpliterator.forEachRemaining (Unknown Source)
at java.util.stream.AbstractPipeline.copyInto (Unknown Source)
at java.util.stream.AbstractPipeline.wrapAndCopyInto (Unknown Source)
at java.util.stream.ForEachOps$ForEachOp.evaluateSequential (Unknown Source)
at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential (Unknown Source)
at java.util.stream.AbstractPipeline.evaluate (Unknown Source)
at java.util.stream.ReferencePipeline.forEach (Unknown Source)
at io.atomix.raft.snapshot.impl.FileBasedSnapshotStore.purgePendingSnapshots (FileBasedSnapshotStore.java:145)
at io.atomix.raft.roles.PassiveRole.abortPendingSnapshots (PassiveRole.java:420)
at io.atomix.raft.roles.PassiveRole.stop (PassiveRole.java:82)
at io.atomix.raft.roles.LeaderRole.stop (LeaderRole.java:119)
at io.atomix.raft.impl.RaftContext.transition (RaftContext.java:501)
at io.atomix.raft.roles.LeaderRole.onVote (LeaderRole.java:581)
at io.atomix.raft.impl.RaftContext.lambda$registerHandlers$16 (RaftContext.java:215)
at io.atomix.raft.impl.RaftContext.lambda$runOnContext$19 (RaftContext.java:224)
at io.atomix.utils.concurrent.SingleThreadContext$WrappedRunnable.run (SingleThreadContext.java:188)
at java.util.concurrent.Executors$RunnableAdapter.call (Unknown Source)
at java.util.concurrent.FutureTask.run (Unknown Source)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker (Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run (Unknown Source)
at java.lang.Thread.run (Unknown Source)
```

</p>
</details>

**Environment:**
 - OS: cloud
 - Zeebe Version: 0.24.4
 - Configuration: -

**Comment by npepinpe** [comment]
Happens seldom (3 times in a month), assumption is that even if you miss a snapshot, you should get the next one, so impact isn't too high, and replication will retry itself (i.e. during startup). To reprioritize if this happens more often or we have any idea of why the impact would be worse.

**Comment by deepthidevaki** [comment]
This should be fixed by #6283
Please reopen if occured again.
